content/blog/2019/egress-traffic-control-in-istio-part-3/index.md

```diff
@@ -0,0 +1,136 @@
+| Istio-aware | {{< cancel_icon >}} | {{< cancel_icon >}} | {{< checkmark_icon >}} |
+
+## Performance considerations
+
+Note that Istio egress control has its price, which is the increased latency of the calls to external services and
+increase of CPU usage by the cluster pods.
+After all, the traffic has to pass through two proxies, namely the sidecar proxy of the
+application and the proxy of the egress gateway. In the case of
+[TLS egress traffic to wildcard domains](/docs/tasks/traffic-management/egress/wildcard-egress-hosts/),
+you must add
+[an additional proxy](/docs/tasks/traffic-management/egress/wildcard-egress-hosts/#wildcard-configuration-for-arbitrary-domains),
+making the count of proxies between the application and the external service three. The traffic between the second and
+third proxies is on the local network of the pod, so it should not have significant impact on the latency.
+
+See a [performance evaluation](/blog/2019/egress-performance/) of different configurations of Istio egress
+traffic control. I would encourage you to measure carefully different configurations for your own applications and your
+external services, and decide whether you can afford the performance overhead for your use cases. You should weigh the
+required level of security versus your performance requirements, and also compare with the performance overhead of
+alternative solutions.
+
+Let me provide our take on the performance overhead of Istio egress traffic control.
+The latency of access to external services could be already high, so adding the overhead
+of two or three proxies inside the cluster could be not very significant.
+After all, in the microservice architecture you can have chains of dozens of calls between microservices, so adding an
+additional hop with one or two proxies in the egress gateway should not have a large impact.
+
+Moreover, we are working to reduce performance overhead of Istio.
+Possible optimizations are to extend Envoy to handle wildcard domains so there will be no need for the
+third proxy; or to use mutual TLS for authentication only, without encrypting the TLS traffic (since it is already
+encrypted).
+
+## Summary
+
+I hope that after reading this series you are convinced that controlling egress traffic is very important for the
+security of your cluster.
+I also hope that I managed to convince you that Istio can serve as an effective tool for controlling egress traffic
+securely, and that Istio has multiple advantages over the alternative solutions.
+In my opinion, you can even choose secure control of egress traffic as the first use case for applying Istio to your
+cluster.
+Istio will already be beneficial for you, even before you start using all other features, such as
+traffic management, security, policies and telemetry, applied to traffic between microservices inside the cluster.
+You should pay attention, however, to performance overhead of Istio egress traffic control and measure it for your use
```

Comment:
I would remove this sentence. It ends the post on a negative note. You already mentioned the performance considerations and I think it would be better to end on the positive note of all the features Istio provides to readers and how secure control of egress traffic can serve as a starting point.

Reply:
Done in https://github.com/istio/istio.io/pull/4637/commits/3c3e8aa3fb3dc258470c6eda7aadcb99eebe22ad